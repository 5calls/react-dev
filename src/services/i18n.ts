import * as i18n from 'i18next';
import * as XHR from 'i18next-xhr-backend';
// import Cache from 'i18next-localstorage-cache';
import * as LanguageDetector from 'i18next-browser-languagedetector';

const instance = i18n
    .use(XHR)
    // .use(Cache)
    .use(LanguageDetector)
    .init({
        backend: {
            loadPath: '/locales/{{lng}}.json'
        },
        fallbackLng: 'en',

        react: {
            wait: true, // globally set to wait for loaded translations in translate hoc
            // exposeNamespace: true // exposes namespace on data-i18next-options to be used in eg. locize-editor
        },

        // have a common namespace used around the full app
        ns: [''],
        defaultNS: '',

        debug: true,

        // cache: {
        //   enabled: true
        // },

        interpolation: {
            escapeValue: false, // not needed for react!!
            formatSeparator: ','
        }
    });

export default instance;
