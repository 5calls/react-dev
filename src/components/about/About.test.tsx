import * as React from 'react';
import { shallow } from 'enzyme';
import i18n from '../../services/i18n';
import { About } from './index';

test('About component snapshot renders correctly', () => {
  const component = shallow(<About t={i18n.t} />);
  expect(component).toMatchSnapshot();
});

test('About component should render', () => {
  const component = shallow(<About t={i18n.t} />);
  const node = component.find('h2');
  expect(node).not.toBeNull();
});