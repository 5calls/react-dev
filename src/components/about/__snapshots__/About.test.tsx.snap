// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component snapshot renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="about"
>
    <h2
        className="about__title"
    />
    <h3
        className="about__subtitle"
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <p />
    <ul>
        <li>
            <span
                dangerouslySetInnerHTML={
                    Object {
                        "__html": undefined,
                      }
                }
            />
        </li>
        <li>
            <span
                dangerouslySetInnerHTML={
                    Object {
                        "__html": undefined,
                      }
                }
            />
        </li>
        <li>
            <span
                dangerouslySetInnerHTML={
                    Object {
                        "__html": undefined,
                      }
                }
            />
        </li>
        <li>
            <span
                dangerouslySetInnerHTML={
                    Object {
                        "__html": undefined,
                      }
                }
            />
        </li>
    </ul>
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <h3
        className="about__subtitle"
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <h3
        className="about__subtitle"
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <h3
        className="about__subtitle"
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <p
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
</section>,
  "nodes": Array [
    <section
      className="about"
>
      <h2
            className="about__title"
      />
      <h3
            className="about__subtitle"
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <p />
      <ul>
            <li>
                  <span
                        dangerouslySetInnerHTML={
                              Object {
                                    "__html": undefined,
                                  }
                        }
                  />
            </li>
            <li>
                  <span
                        dangerouslySetInnerHTML={
                              Object {
                                    "__html": undefined,
                                  }
                        }
                  />
            </li>
            <li>
                  <span
                        dangerouslySetInnerHTML={
                              Object {
                                    "__html": undefined,
                                  }
                        }
                  />
            </li>
            <li>
                  <span
                        dangerouslySetInnerHTML={
                              Object {
                                    "__html": undefined,
                                  }
                        }
                  />
            </li>
      </ul>
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <h3
            className="about__subtitle"
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <h3
            className="about__subtitle"
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <h3
            className="about__subtitle"
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <p
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        t={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "t": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="about"
>
          <h2
                    className="about__title"
          />
          <h3
                    className="about__subtitle"
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p />
          <ul>
                    <li>
                              <span
                                        dangerouslySetInnerHTML={
                                                  Object {
                                                            "__html": undefined,
                                                          }
                                        }
                              />
                    </li>
                    <li>
                              <span
                                        dangerouslySetInnerHTML={
                                                  Object {
                                                            "__html": undefined,
                                                          }
                                        }
                              />
                    </li>
                    <li>
                              <span
                                        dangerouslySetInnerHTML={
                                                  Object {
                                                            "__html": undefined,
                                                          }
                                        }
                              />
                    </li>
                    <li>
                              <span
                                        dangerouslySetInnerHTML={
                                                  Object {
                                                            "__html": undefined,
                                                          }
                                        }
                              />
                    </li>
          </ul>
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <h3
                    className="about__subtitle"
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <h3
                    className="about__subtitle"
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <h3
                    className="about__subtitle"
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="about"
>
          <h2
                    className="about__title"
          />
          <h3
                    className="about__subtitle"
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p />
          <ul>
                    <li>
                              <span
                                        dangerouslySetInnerHTML={
                                                  Object {
                                                            "__html": undefined,
                                                          }
                                        }
                              />
                    </li>
                    <li>
                              <span
                                        dangerouslySetInnerHTML={
                                                  Object {
                                                            "__html": undefined,
                                                          }
                                        }
                              />
                    </li>
                    <li>
                              <span
                                        dangerouslySetInnerHTML={
                                                  Object {
                                                            "__html": undefined,
                                                          }
                                        }
                              />
                    </li>
                    <li>
                              <span
                                        dangerouslySetInnerHTML={
                                                  Object {
                                                            "__html": undefined,
                                                          }
                                        }
                              />
                    </li>
          </ul>
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <h3
                    className="about__subtitle"
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <h3
                    className="about__subtitle"
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <h3
                    className="about__subtitle"
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    t={[Function]}
/>,
}
`;
